### ISSUE TITLE
Windows auto builds don't terminate when an external process fails

### ISSUE THREAD

**Issue by oblivioncth** [oblivioncth/Qx#20]
Issue with PS. Current solution would require adding last error code check to every external process start.

Instead, waiting on feature new to have this behavior be automatic in 7.3.0

https://github.com/PowerShell/PowerShell-RFC/pull/277
https://docs.microsoft.com/en-us/powershell/scripting/whats-new/what-s-new-in-powershell-73?view=powershell-7.2

**Comment by oblivioncth** [comment]
The feature is now available in 7.3.0 as an experimental feature. This means that in order to use it, one must execute the following and then restart Powershell:
`Enable-ExperimentalFeature PSNativeCommandErrorActionPreference`

This means that every workflow job would need to have an extra step that does the above, which would be quite inconvenient, especially given that the jobs also need a step for enabling the feature on that specific "script":
`$PSNativeCommandUseErrorActionPreference = $true`.

Probably will wait until the feature isn't experimental unless there's an easier way to enable experimental features in PS for workflows globally. Additionally, research needs to be done to see if there is a similarly easy way to set the preference itself globally.

**Comment by oblivioncth** [comment]
Starting with 7.4.0 `$PSNativeCommandUseErrorActionPreference = $true` is the default, so currently the only remaining blocker is that the feature is still experimental and therefore cannot be activated conveniently for workflows.

**Comment by oblivioncth** [comment]
Looks like it actually went mainstream with 7.4.0 so I just need to add it to the script and test that it works.

**Comment by oblivioncth** [comment]
I believe we just want:
```
$DebugPreference = "Stop"
$PSNativeCommandUseErrorActionPreference = $true
```
At the top of applicable pwsh scripts.

**Comment by oblivioncth** [comment]
Seems like in theory GitHub should be open to enabling this by default based on their existing default settings for pwsh and bash that prefer "fail-fast" behavior. This was addressed once before, but left alone as it was believed that pwsh 7.4 was going to have the feature enabled by default anyway; however, that ended up not being the case:

https://github.com/actions/runner-images/issues/8900

Probably should make a new issue if this point is ignored for a while.
